src/System.Private.CoreLib/src/Internal/Diagnostics/Tracing/FrameworkEventSourceSupport.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+
+namespace Internal.Diagnostics.Tracing
+{
+    //
+    // An internal contract that exposes just enough async debugger support needed by the AsTask() extension methods in the WindowsRuntimeSystemExtensions class.
+    //
+    public static class FrameworkEventSourceSupport
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsEnabled(EventLevel level, EventKeywords keywords)
+        {
+            return FrameworkEventSource.Log.IsEnabled(level, keywords);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ThreadTransferSendObj(object id, int kind, string info, bool multiDequeues)
```

Comment:
@brianrob @vancem Do you believe that this `ThreadTransfer` ETW tracing for WinRT is useful?

We do not have it in ProjectN and nobody missed it in last 5 years...

Reply:
My question was specifically about the events we have around the WinRT CoreDispatcher:
https://github.com/dotnet/corefx/blob/master/src/System.Runtime.WindowsRuntime/src/System/Threading/WindowsRuntimeSynchronizationContext.cs#L184

Everything in WinRT OS implementation is instrumented via the (WinRT causality tracing)[https://docs.microsoft.com/en-us/uwp/api/Windows.Foundation.Diagnostics.AsyncCausalityTracer], including the WinRT CoreDispatcher.

I would think that perfview should be able to do better stack stitching by consuming the WinRT causality tracing ETW events because of they are more complete. Or is there a reason why that does not work?

Reply:
I don't have first-hand information on your question.   PerfView does have generic logic that will use ThreadTransferSend and ThreadTransferReceive to stitch together causality, and so in theory  the wiring of this into WinRTs causalityTracer should 'just work' and be beneficial.   But I don't have first hand experience seeing the results of this.    I also don't know in what way (or how much) Visual Studio takes advantage of this information.   

Presumably this in references to @stephentoub s question of who uses the events and whether we could change how we emit them.    PerfView shuld be using  these events, but clearly we do not have tests to insure that they actually work end-to-end.  

Do we have a handy UWP app that we can run a 'perfView /threadTime collect' with the updated changes?   I can see that the events are there and what PerfView does with them.